### PR TITLE
Raw method

### PR DESCRIPTION
**Pull request by fxmanu**

### CODE REVIEW

File: src/billomat-resource-client.ts [hunk 1]
```diff
@@ -92,6 +92,35 @@ export class BillomatResourceClient<T extends Billomat.Resource> {
         });
     }
 
+    public raw<TResult extends object>(method: string, subUri: string, payload?: object | null, query?: { [key: string]: string }): Promise<TResult> {
```
[the remark this answers is not in the document]
Comment (j-frost) [reply src/billomat-resource-client.ts:95]: To be more in line with the rest of the code base (and pass the linter): 

```typescript
    public raw<TResult extends unknown>(
        method: string,
        subUri?: string,
        options?: Billomat.RawOptions
    ): Promise<TResult> {
        return new Promise((resolve, reject) => {
            const singular = SINGULAR.get(this._name);
            if (singular === undefined) {
                reject('Unsupported resource (no singular defined)');
                return;
            }

            const uri = `api/${this._name}${subUri ? `/${subUri}` : ''}`;
            let req = this.createAuthedRequest(method, uri).query(options?.query ?? {});

            if (options?.payload) {
                req = req.send(options.payload);
            }

            req.then((response) => {
                if (!this.isRawResponse(response.body)) {
                    reject(`Invalid response: ${JSON.stringify(response.body)}`);
                    return;
                }
                resolve(response.body as TResult);
            }).catch(reject);
        });
    }
// ...
    private isRawResponse(o: unknown): o is Record<string, T> {
        return typeof o === 'object' && o !== null;
    }
```

Comment (j-frost) [inline src/billomat-resource-client.ts:95]: Also strongly feel this should have a sanity check test like so:

```typescript
    describe('when calling raw', () => {
        beforeEach(() => scope.get(new RegExp('api/invoices/1')).reply(200, { ok: 'yes' }));

        it('returns response body', (done: Mocha.Done) => {
            api['invoices']
                .raw('GET', '1')
                .then((response) => {
                    expect(response).to.deep.equal({ ok: 'yes' });
                    done();
                })
                .catch((error) => done(error));
        });
    });
```

**Review by j-frost — COMMENTED** [review]
If my ideas for changes sound good to you I have them ready to go @fxmanu

**Comment by j-frost** [comment]
I've pushed what I'd suggest to do to here https://github.com/fxmanu/repo/compare/raw-method...seibert-media:raw

File: src/billomat-resource-client.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fxmanu) [reply src/billomat-resource-client.ts:95]: @j-frost an options parameter is a good idea 👍

**Comment by fxmanu** [comment]
@j-frost looks good to me. Thanks for your contribution :-)

**Pull request merged**